python/tk_alias/menu_generation.py

```diff
@@ -186,6 +188,23 @@ def _jump_to_fs(self):
             if exit_code != 0:
                 self._engine.logger.error("Failed to launch '%s'!", cmd)
 
+    def _version_check(self, version):
+        """
+        Check if Alias version is greater than 2022.2 for ShotGrid menu name
+
+        :param version: String value of Alias version from startup.py
+        """
+        multiply_by_10 = False
+        if len(version) == 4:
+            multiply_by_10 = True
+        elif len(version) > 6:
+            version = version[0:5]
+        version_int = int(version.replace(".", ""))
+        if multiply_by_10:
+            version_int = version_int * 10
+
+        return version_int
+
```

Comment:
This function doesn't actually check if the version is greater than 2022.2 (as in the function comment), it processes the version and returns the it as an int.

I would suggest replacing this function with `_version_compare` which takes two versions and compares them by going through each part of the version, `MAJOR.MINOR.PATCH`, like so:

```
# Method to compare two versions.
# Return 1 if v2 is smaller,
# -1 if v1 is smaller,,
# 0 if equal
def versionCompare(v1, v2):
     
    # This will split both the versions by '.'
    arr1 = v1.split(".")
    arr2 = v2.split(".")
    n = len(arr1)
    m = len(arr2)
     
    # converts to integer from string
    arr1 = [int(i) for i in arr1]
    arr2 = [int(i) for i in arr2]
  
    # compares which list is bigger and fills
    # smaller list with zero (for unequal delimeters)
    if n>m:
      for i in range(m, n):
         arr2.append(0)
    elif m>n:
      for i in range(n, m):
         arr1.append(0)
     
    # returns 1 if version 1 is bigger and -1 if
    # version 2 is bigger and 0 if equal
    for i in range(len(arr1)):
      if arr1[i]>arr2[i]:
         return 1
      elif arr2[i]>arr1[i]:
         return -1
    return 0
```

(For reference: https://www.geeksforgeeks.org/compare-two-version-numbers/)

Then line 32 would become:
```
if self._version_compare(engine.alias_version, "2022.2") >= 0:
```